ShiftNrg/shift-explorer
# Prefer exact delegate matches

### What was the problem?

If searching for a delegate by username, it may return the incorrect username if it appears alphabetically before the exact username.

e.g. Currently searching for `wariner` returns `pool_wariner2_voting`

### How did I fix it?

First perform an exact username search. If that fails then fallback to a regular search. Tried to follow the same code-style/structure as the rest of the file

### How to test it?

I unfortunately wasn't able to get the testing DB synced and working as per the instructions. I would have liked to add another delegate to the testing database (with a name that appears alphabetically before the others, i.e. `genesis`, to see if I can search for `gene` (to return `genesis_1`, and search for `genesis` to return `genesis` as expected)

i.e.
```js
describe('GET /api/getSearch', () => {
  ...
  it('should prefer exact matches', (done) => { ... });
});
```

I have checked it manually for the following cases:
- Search partial name still works: e.g. `gen` -> `genesisDelgate1`
- Search exact name now works when there is another delegate match: e.g. `wariner` -> `wariner` (previously `pool_wariner_2`), `loomdart` -> `loomdart` (previously `inloomdartwetrust`), etc
- Search by address still works in explorer
